src/hooks/useTheme.ts

```diff
@@ -0,0 +1,17 @@
+import usePersistedState from './usePersistedState'
+
+import { DarkTheme, LightTheme } from '~styles/themes'
+
+const useTheme = () => {
+  const [themeMode, setThemeMode] = usePersistedState<'light' | 'dark'>('theme', 'dark')
+
+  const toggleTheme = (): void => {
+    setThemeMode((themeMode === 'light' ? 'dark' : 'light') as React.SetStateAction<'light' | 'dark'>)
```

Comment:
Pode remover o cast de tipagem 

```suggestion
    const switchTheme = themeMode === 'light' ? 'dark' : 'light';
    
    setThemeMode(switchTheme)
```